### PR TITLE
lockfile build-order working for consumers too

### LINKED ISSUE #5727: `ERROR: Cannot get the latest package without RREV` from `conan graph build-order` when all packages are cached
Conan version: **1.18.2**

I'm designing our new CI buid process based on lockfiles and revisions. If I have all packages built in the local cache and rerun the build, I'm getting the error mentioned in the the title. This is the invocation:

```
❯ ./conan-bin graph build-order --build cascade --build outdated conan.lock
Using lockfile: '/home/tamas/work/conan/conan.lock'
bzip2/1.0.6: Package is up to date
iconv/1.15: Package is up to date
zlib/1.2.8: Package is up to date
boost/1.69.0: Package is up to date
ERROR: Cannot get the latest package without RREV. [Remote: toolchain]
```

If I remove at least one of the packages that were built (via `conan remove`) I'm not getting an error.

I have no idea what this means, any clues? What can I do to avoid it? 

Also, I think `CONAN_VERBOSE_TRACEBACK` is broken as I'm not able to get a traceback for this error.

EDIT: I created a minimal(-ish) repro [here](https://github.com/sztomi/cannot_get_latest_package). See the README in that repo for instructions.

### PR DESCRIPTION
**Pull request by memsharded**
Changelog:  Bugfix: A lockfile generated from a consumer should be able to generate a build-order too.
Docs: Omit

Fix #5727

#tags: slow

### CODE REVIEW

**Comment by memsharded** [comment]
@lasote would like to discuss a couple of rough edges over locking build-requires

File: conans/client/graph/graph.py [hunk 1]
```diff
@@ -284,7 +284,7 @@ def new_build_order(self):
             new_level = []
             for n in level:
                 if n.binary == BINARY_BUILD and n.pref not in total_prefs:
-                    new_level.append((n.id, repr(n.pref)))
+                    new_level.append((n.id, n.pref.copy_clear_rev()))
```
Comment (lasote) [inline conans/client/graph/graph.py]: `copy_clear_revs()` instead? why you want to keep the `rrev` and not the `prev`?

File: conans/model/graph_lock.py [hunk 2]
```diff
@@ -137,7 +145,12 @@ def root_node(self):
             total.extend(node.requires.values())
         roots = set(self._nodes).difference(total)
         assert len(roots) == 1
-        return self._nodes[roots.pop()]
+        root_node = self._nodes[roots.pop()]
+        if root_node.path:
+            return root_node.path
+        if not self.revisions_enabled:
+            return root_node.pref.ref.copy_clear_rev()
```
Comment (lasote) [inline conans/model/graph_lock.py]: Idem about `copy_clear_revs()`

File: conans/client/graph/graph.py (hunk 1 above)
Comment (memsharded) [reply conans/client/graph/graph.py]: No, we certainly want to build a revision, the build order will be building recipes with a given revision, that shouldn't change.

**Pull request merged**